admin/class-metabox.php

```diff
@@ -936,1318 +617,51 @@ public function localize_media_script() {
-
-		unset( $results );
-
-		return $output;
+		return $GLOBALS['post'];
 	}
 
 	/**
-	 * Calculate the page analysis results for post.
+	 * Counting the number of given keyword used for other posts than given post_id
 	 *
-	 * @todo [JRF => whomever] check whether the results of this method are always checked with is_wp_error()
-	 * @todo [JRF => whomever] check the usage of this method as it's quite intense/heavy, see if it's only
-	 * used when really necessary
-	 * @todo [JRF => whomever] see if we can get rid of the passing by reference of $results as it makes
-	 * the code obfuscated
+	 * @param integer $post_id
 	 *
-	 * @param  object $post Post to calculate the results for.
-	 *
-	 * @return  array|WP_Error
+	 * @return int
 	 */
-	function calculate_results( $post ) {
-		$options = WPSEO_Options::get_all();
-
-		if ( ! class_exists( 'DOMDocument' ) ) {
-			$result = new WP_Error( 'no-domdocument', sprintf( __( "Your hosting environment does not support PHP's %sDocument Object Model%s.", 'wordpress-seo' ), '<a href="http://php.net/manual/en/book.dom.php">', '</a>' ) . ' ' . __( "To enjoy all the benefits of the page analysis feature, you'll need to (get your host to) install it.", 'wordpress-seo' ) );
-
-			return $result;
-		}
-
-		if ( ! is_array( $post ) && ! is_object( $post ) ) {
-			$result = new WP_Error( 'no-post', __( 'No post content to analyse.', 'wordpress-seo' ) );
-
-			return $result;
-		}
-		elseif ( self::get_value( 'focuskw', $post->ID ) === '' ) {
-			$result = new WP_Error( 'no-focuskw', sprintf( __( 'No focus keyword was set for this %s. If you do not set a focus keyword, no score can be calculated.', 'wordpress-seo' ), $post->post_type ) );
-
-			self::set_value( 'linkdex', 0, $post->ID );
-
-			return $result;
-		}
-		elseif ( apply_filters( 'wpseo_use_page_analysis', true ) !== true ) {
-			$result = new WP_Error( 'page-analysis-disabled', sprintf( __( 'Page Analysis has been disabled.', 'wordpress-seo' ), $post->post_type ) );
-
-			return $result;
-		}
-
-		$results = array();
-		$job     = array();
-
-		$sampleurl             = $this->get_sample_permalink( $post );
-		$job['pageUrl']        = preg_replace( '`%(?:post|page)name%`', $sampleurl[1], $sampleurl[0] );
-		$job['pageSlug']       = urldecode( $post->post_name );
-		$job['keyword']        = self::get_value( 'focuskw', $post->ID );
-		$job['keyword_folded'] = $this->strip_separators_and_fold( $job['keyword'] );
-		$job['post_id']        = $post->ID;
-		$job['post_type']      = $post->post_type;
-
-		$dom                      = new domDocument;
-		$dom->strictErrorChecking = false;
-		$dom->preserveWhiteSpace  = false;
-
-		/**
-		 * Filter: 'wpseo_pre_analysis_post_content' - Make the post content filterable before calculating the page analysis
-		 *
-		 * @api string $post_content The post content
-		 *
-		 * @param object $post The post.
-		 */
-		$post_content = apply_filters( 'wpseo_pre_analysis_post_content', $post->post_content, $post );
```

Comment:
This one is for instance found over 6000 times in GitHub search, see https://github.com/search?q=wpseo_pre_analysis_post_content&type=Code&utf8=%E2%9C%93